apache/superset
# fix(AlertReportModal): Text Area Change

### SUMMARY

There was an error report where users were reporting that sql values that they were inputting into the TextAreaControl section would linger even when they closed out and went to a new report, or even created a new report. This was because we were storing the sql in state, but not clearing state when a modal was closed. 

This state was created to stop cursor errors here: https://github.com/apache/superset/pull/15820. I believe that @geido made this change. 

The cursor errors were occurring because we added a debouncer to the ace-editor here: https://github.com/apache/superset/pull/10837

In investigating this, I was wondering why we had a debouncer for the ace editor to begin with? The input is not using an API, and looking through the different places where TextAreaControl is used, none of them make an API call. So this PR deletes the debouncer, which fixes both the cursor back issue and also gets rid of the state.value //prop.value difference. 

However, I wanted to make sure that getting rid of this debouncer is fine @ktmud and @geido ? Where are the places that we should check to make sure?

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF


Before:


https://user-images.githubusercontent.com/48933336/138182082-471e1948-0591-4f86-a4b8-21c264a41ff3.mp4


After:
https://user-images.githubusercontent.com/48933336/138182251-f5d5dfc0-c7dc-4066-a39a-b1b2b1b0cffd.mov

### TESTING INSTRUCTIONS
Create an alert, exit out. Create another alert, there should not be any sql in the sql input form. 

### ADDITIONAL INFORMATION
